Courier's Helpers MK2, again

courierbravo opened the pull request:
God help me.

PR does the following:

- Adds the Apollo event shuttle we've been using, along with 2 variants of it, and several other copy/paste-able event shuttles to use as start points. Also, fluff writing regarding it that's subject to change. 
- Adds a pipes guideline to give some guidance on preferred ways of doing scrubber/supply pipes. Also copy paste ready atmos stuff.

- Adds a ton of industrial flooring decal color presets.

- Refactors /obj/effect/floor_decal/industrial/outline/engineering into outline/engineering and outline/operations. Replaces all instances of outline/engineering with outline/operations. This was done because outline/engineering was using brown as its color, which isn't really an engineering color.

courierbravo commented:
!review

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/e1f50c85-5cd5-470d-9b64-a3e945e7a2ca)
folder should be `event_shuttles

DreamySkrell commented on code/game/turfs/flooring/flooring_decals_industrial.dm:
```diff
@@ -28,60 +28,249 @@
 
+/obj/effect/floor_decal/industrial/hatch/blue
+	name = "cyan hatched marking"
+	color = "#b6efe1"
```
should use one of the color defines, like COLOR_BLUE_LIGHT or something (idk I feel `"#b6efe1"` is a bit of a boring color)

also the name should say "blue" to match the type

this applies to all of the /blue decals here

courierbravo replied:
It was already like that when i started, see: outlines. It’s not something I did, I’m just copying it because the color gets used. Can fix in a follow up PR.

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/02b1cb6e-5f73-4393-a4fb-9c2afa027676)
bad `level` var set on some of these vents

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/36087af4-faab-4847-a992-94d9e3ae1e5d)
do not use objects with invalid sprites (exception is stuff like spawners) - it looks like it may be broken on spawn, even if it spawns correctly

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/91420281-7bc2-4046-8aa1-ea3982754409)
not sure if a crate like this is going to close after being opened, spilling items on the floor

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/4e19c5c5-f4a8-4c87-b94e-05f9985ff2c6)
the shuttles should have this area mapped in, instead of just being the space area

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/14c799fe-d493-47a5-a135-d1247bd4826c)
don't call it a meme
I think the idea for the shuttle is fine, but calling it a meme may make someone just disregard it as unserious

courierbravo commented:
> ![image](https://private-user-images.githubusercontent.com/107256943/353618568-36087af4-faab-4847-a992-94d9e3ae1e5d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MzI0NDAsIm5iZiI6MTcyMjQzMjE0MCwicGF0aCI6Ii8xMDcyNTY5NDMvMzUzNjE4NTY4LTM2MDg3YWY0LWZhYWItNDg0Ny1hOTkyLTk0ZDllM2FlMWU1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMVQxMzIyMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MmRmMWU4MDNiZmZmYWIyYjAxYzgwMDUzMGQ1YzU4YWE3MzA3OTQ4ZjdhMmQ1YTBkZDhkNWE5Y2E0MzBjODhiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.j6-vdx2pfoGs9oQVGkZ7a0Gt8DHv6e0_VO_Mg2OSsPQ)
> do not use objects with invalid sprites (exception is stuff like spawners) - it looks like it may be broken on spawn, even if it spawns correctly

These things break regularly, i don’t think it’s fair to point it out on my PR when I’m just trying to utilize the item. 



> ![image](https://private-user-images.githubusercontent.com/107256943/353619110-91420281-7bc2-4046-8aa1-ea3982754409.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MzI0NDAsIm5iZiI6MTcyMjQzMjE0MCwicGF0aCI6Ii8xMDcyNTY5NDMvMzUzNjE5MTEwLTkxNDIwMjgxLTdiYzItNDA0Ni04YWExLWVhMzk4Mjc1NDQwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMVQxMzIyMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMTY1ZTM4MDBmNThmMzU1ZDJiNWEwNTdkYjE1ZTc2YTg2ZGYyNzExNDA1NzEzOGQzMGI2MTAwNWYxYzYyODY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FWNNDktE71LQF1pb2Y3T-e-3c7jWkLzfu4nzJLPccxE)
> not sure if a crate like this is going to close after being opened, spilling items on the floor
> better would be a crate of boxes with different encryption keys, or even a vendor
> also a ton easier for people to take out the keys they want from this, without getting a ton of lag right-clicking a pile of so many items

That’s fair, but outside the scope of this PR. The crate does close, at least. 



> ![image](https://private-user-images.githubusercontent.com/107256943/353621112-76b05a42-50e3-4cf7-b5aa-c7c49392c4ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MzI0NDAsIm5iZiI6MTcyMjQzMjE0MCwicGF0aCI6Ii8xMDcyNTY5NDMvMzUzNjIxMTEyLTc2YjA1YTQyLTUwZTMtNGNmNy1iNWFhLWM3YzQ5MzkyYzRhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMVQxMzIyMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzA1MTFmZWNiMDBhOTVmNTYzN2NiY2UyMzhmYTVlNWE3MDNmYmQ3OTM5YjE2YzE2OGQxN2MwZDRiYTFiZjk0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ax8YTcgNxEPB-eF_FeYPkuohCmyRh4WVK98Y5NQhCaM) I don't think we're going to have any future events with a ton of weapons for security and crew militia, like the konyang events especially events that require such a strong gear loadout - tons of heavy armor sets, tons of rifles, etc
> 
> I don't think the artemis shuttle is going to be used anytime soon, so I see little point in including it here

I don’t think there’s any harm in including it either. I can see the Artemis being useful for something like another zombies event. My vision for these helpers isn’t just to provide a standardized set of useful shuttles, but to also inspire their use for both canon and non canon events. 

All unreplied comments will be addressed in simple fixes.

DreamySkrell commented:
(replied to the above on discord)

courierbravo commented:
Almost all requested changes made or were discussed
-fixed name of folder 
-fixed level variables
-fixed icon state issue
-mapped in areas

To Do in follow up PRs:
Make a vendor for encryption keys (starting today 07/31/24)
Refactor applicable markings in the industrial flooring decals (create variable for the color as well instead of using hex)
